crates/bevy-inspector-egui/src/bevy_inspector/mod.rs

```diff
@@ -239,13 +239,28 @@ pub fn ui_for_state<T: FreelyMutableState + Reflect>(world: &mut World, ui: &mut
 
 /// Display all entities and their components
 pub fn ui_for_world_entities(world: &mut World, ui: &mut egui::Ui) {
-    ui_for_world_entities_filtered::<Without<Parent>>(world, ui, true);
+    ui_for_world_entities_filtered_from_ui::<Without<Parent>>(world, ui, true);
+}
+
+pub trait EntityFilter {
+    /// Returns true if the filter term is currently empty
+    ///
+    /// default impl is false
+    fn is_empty(&self) -> bool {
+        false
+    }
+    /// Returns [`Vec<Entity>`] filtered by the filter term
+    fn filtered_entities(
+        &self,
+        world: &mut World,
+        entities: impl IntoIterator<Item = Entity>,
+    ) -> Vec<Entity>;
 }
 
 #[derive(Debug, Clone)]
-struct Filter {
-    word: String,
-    is_fuzzy: bool,
+pub struct Filter {
+    pub word: String,
+    pub is_fuzzy: bool,
```

Comment:
I just realized that `Filter` doesn't have to be pub now that I've added a trait. I can revert it if you want to prevent it from being exposed or add unnecessary additional support requirements.